sql/handler.cc

```diff
@@ -2145,14 +2149,16 @@ commit_one_phase_2(THD *thd, bool all, THD_TRANS *trans, bool is_real_trans)
 
   if (ha_info)
   {
-    int err;
+    int err= 0;
 
     if (has_binlog_hton(ha_info) &&
         (err= binlog_commit(thd, all,
                             is_ro_1pc_trans(thd, ha_info, all, is_real_trans))))
     {
       my_error(ER_ERROR_DURING_COMMIT, MYF(0), err);
       error= 1;
+
+      goto err;
```

Comment:
okay, this actually makes sense.

Although I'm not sure on what code path we can get here (in `commit_one_phase_2()` with `has_binlog_hton()` being non-zero and `binlog_commit()` being non-trivial)

But let's have it just in case, sure

Reply:
>  what code path we can get here

[binlog_commit_fail.test](https://github.com/MariaDB/server/blob/bb-10.6-MDEV-35506/mysql-test/suite/binlog/t/binlog_commit_fail.test#L73`) provides one. And [also](https://github.com/MariaDB/server/blob/bb-10.6-MDEV-35506/mysql-test/suite/binlog/t/binlog_commit_fail.test#L60).